src/kubectlutils.ts

```diff
@@ -17,157 +17,126 @@
 import * as k8s from 'vscode-kubernetes-tools-api';
 import * as config from './config';
 
-export function isKubernetesAvailable(): Promise<boolean> {
-	return new Promise<boolean>( (resolve) => {
-		k8s.extension.kubectl.v1
-			.then( (kubectl) => {
-				resolve(kubectl && kubectl.available);
-			})
-			.catch( err => {
-				resolve(false);
-			});
-	});
+export async function isKubernetesAvailable(): Promise<boolean> {
+	const kubectl: k8s.API<k8s.KubectlV1> = await k8s.extension.kubectl.v1;
+	return kubectl && kubectl.available;
 }
 
-export function getNamedListFromKubernetes(itemType : string, extra? : string): Promise<string> {
-	return new Promise<string>( async (resolve, reject) => {
-		await k8s.extension.kubectl.v1
-			.then( async (kubectl) => {
-				let cmd = `get ${itemType}`;
-				if (extra) {
-					cmd += ` ${extra}`;
-				}
-				if (kubectl && kubectl.available) {
-					return await kubectl.api.invokeCommand(cmd);
-				} else {
-					reject(new Error('Kubernetes not available'));
-				}
-			})
-			.then( (result) => {
-				if (!result || result.code !== 0) {
-					let error = `Unable to invoke kubectl to retrieve ${itemType}`;
-					if (result && result.stderr) {
-						error = result.stderr;
-					}  
-					reject(error);
-				} else if (result) {
-					const splitResults = result.stdout;
-					resolve(splitResults);
-				}
-			})
-			.catch( (err) => reject(err) );
-	});
+export async function getNamedListFromKubernetes(itemType : string, extra? : string): Promise<string> {
+	const kubectl: k8s.API<k8s.KubectlV1> = await k8s.extension.kubectl.v1;
+	let cmd = `get ${itemType}`;
+	if (extra) {
+		cmd += ` ${extra}`;
+	}
+	if (kubectl && kubectl.available) {
+		const result: k8s.KubectlV1.ShellResult | undefined = await kubectl.api.invokeCommand(cmd);
+		if (!result || result.code !== 0) {
+			let error = `Unable to invoke kubectl to retrieve ${itemType}`;
+			if (result && result.stderr) {
+				error = result.stderr;
+			}  
+			return error;
+		} else if (result) {
+			const splitResults = result.stdout;
+			return splitResults;
+		}
+	}
+	return 'Kubernetes not available';
 }
 
-export function getNamedListFromKubernetesThenParseList(itemType : string, extra? : string): Promise<string[]> {
-	return new Promise<string[]>( (resolve, reject) => {
-		getNamedListFromKubernetes(itemType, extra)
-			.then ((result) => {
-				const itemList : string[] = parseShellResult(result);
-				resolve(itemList);
-				return;
-			}).catch( (error) => {
-				reject(error);
-				return;
-			});
-	});
+export async function getNamedListFromKubernetesThenParseList(itemType : string, extra? : string): Promise<string[]> {
+	try {
+		const result = await getNamedListFromKubernetes(itemType, extra);
+		const itemList : string[] = parseShellResult(result);
+		return itemList;
+	} catch (error) {
+		return [error];
+	}
```

Comment:
aren't we losing the information that there is an error? the array of string would seem like there is one elements returned instead of an error?

Reply:
yeah, that is probably right. I will change that.

Reply:
Yes. I tend to agree. The catch before was...

```
			}).catch( (error) => {
				reject(error);
				return;
			});
```

And the error was at least bubbled up the chain. Maybe if we change the return to Promise<string[]|any> and check the return when we get something back from this method?